### PR TITLE
3771 provider backfill

### PR DESCRIPTION
**Pull request by kpethtel**
## Description of change
When provider name is not provided in upstream index, attempt to set appointment provider name by fetching it from providers service. This is consistent with web behavior.

## Original issue(s)
department-of-veterans-affairs/va-mobile-team#3771

## Things to know about this PR
Will test on review instance.

### CODE REVIEW

File: modules/mobile/app/services/mobile/v2/appointments/provider_name_aggregator.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
```
Comment (kpethtel) [inline modules/mobile/app/services/mobile/v2/appointments/provider_name_aggregator.rb:1]: Keeping this logic in a service object because the proxy was getting very busy and because the specs for this are growing harder to write as we add more external calls, which is making the VCR cassettes difficult to work with. Extracting into a service object makes it easier to write extensive unit tests that cover edge cases. The hard part was determining where it should go and what it should be called. I decided to leave it in the folder alongside the proxy because it's pretty specific to the proxy. And, despite conversations I've had recently with @Tonksthebear about using "model" like names, I had to go with an action name here because nothing about this feels like a model to me.

File: modules/mobile/spec/request/appointments_vaos_v2_list_request_spec.rb [hunk 2]
```diff
@@ -210,5 +214,74 @@
...
+          VCR.use_cassette('providers/get_provider_200', match_requests_on: %i[method uri], tag: :force_utf8) do
+            get '/mobile/v0/appointments', headers: iam_headers, params: params
+          end
+        end
+      end
+
+      context 'when upstream appointments index returns provider names' do
+        it 'adds names to healthcareProvider field' do
+          fetch_appointments
+          # it would be better to find by id, but there's currently a bug that's causing these ids to be nil
```
Comment (kpethtel) [inline modules/mobile/spec/request/appointments_vaos_v2_list_request_spec.rb:239]: When andrew recorded the "all of ruben's appointments" vcr, many of the appointments didn't have ids. It seems that they do now. After this ticket, I will check staging and update the VCRs if they're substantially different, which will hopefully reveal any issues in the proxy as a result of those data changes.

File: modules/mobile/spec/support/vcr_cassettes/appointments/VAOS_v2/get_appointments_with_mixed_provider_types.yml [hunk 3]
```diff
@@ -0,0 +1,122 @@
+---
```
Comment (kpethtel) [inline modules/mobile/spec/support/vcr_cassettes/appointments/VAOS_v2/get_appointments_with_mixed_provider_types.yml:1]: I created this VCR by copying a few appointments from the ruben vcr and modifying them. Not ideal, but none of the appointments in that VCR fit some of these test cases.

File: modules/mobile/spec/support/vcr_cassettes/providers/get_provider_200.yml [hunk 4]
```diff
@@ -0,0 +1,238 @@
+---
```
Comment (kpethtel) [inline modules/mobile/spec/support/vcr_cassettes/providers/get_provider_200.yml:1]: This was copied from the VAOS cassettes and modified. I needed to add a few different provider ids to cover all the cases in ruben's appointments data, so you'll see that there are four requests here and the only differences are the provider ids.

File: modules/mobile/spec/services/v2/appointments/provider_names_spec.rb [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+            "value": '520647669'
+          }
+        ],
+        "name": {
+          "family": 'KNIEFEL',
+          "given": [
+            'CAROLYN'
+          ]
+        },
+        "practiceName": 'Cheyenne WY VAMC'
```
Comment (aherzberg) [inline modules/mobile/spec/services/v2/appointments/provider_names_spec.rb:23]: where is this practiceName coming from? I'm not seeing it on VAOS v2 docs. was it from a VCR?

**Review by aherzberg — APPROVED** [review]
LGTM. very robust testing.

File: modules/mobile/spec/services/v2/appointments/provider_names_spec.rb (hunk 5 above)
Comment (kpethtel) [reply modules/mobile/spec/services/v2/appointments/provider_names_spec.rb]: I just removed this since it adds nothing to the test.

**Review by aherzberg — APPROVED** [review]
LGTM
